src/leo-server-app/src/routes/satellite.ts

```diff
@@ -224,6 +247,87 @@ router.get("/getNextPasses", (req: any, res: any) => {
   }
 });
 
+router.get("/getSolarIlluminationCycle", (req: any, res: any) => {
+  try {
+    // Time window in milliseconds (1 minute)
+    const WINDOWMILLIS = 10 * 1000;
+    // Minimum duration for illumination cycle in milliseconds (10 minutes)
+    const MIN_CYCLE_DURATION = 10 * 60 * 1000;
+
+    var today = new Date();
+    today.setHours(0, 0, 0, 0);
+
+    var nextIlluminations = [];
+    var enterIllumination = null;
+    var enterTime = null;
+    var enterInfo;
+    var exitInfo;
+
+    // Gets illuminations for the next week
+    for (let i = 0; i < 7 * 24 * 60; i++) {
+      // Calculate the next illumination
+      var nextIlluminationTime = new Date(today.getTime() + i * WINDOWMILLIS);
+
+      // Get satellite information for the next pass
+      var satelliteInfo = getSatelliteInfo(
+        nextIlluminationTime,
+        tleLine1,
+        tleLine2
+      );
+      let longitude = satelliteInfo.longitude;
+      let latitude = satelliteInfo.latitude;
+      let height = satelliteInfo.height;
+
+      // Format Time
+      const formattedTime = nextIlluminationTime
+        .toLocaleString("en-US", {
+          year: "numeric",
+          month: "short",
+          day: "2-digit",
+          hour: "2-digit",
+          minute: "2-digit",
+          second: "2-digit",
+        })
+        .replace(/\u202f/g, " ");
+
+      if (isSunlit(nextIlluminationTime, longitude, latitude, height)) {
+        if (!enterIllumination) {
+          enterInfo = {
+            type: "Enter",
+            time: formattedTime,
+            longitude: longitude,
+            latitude: latitude,
+          };
+          enterIllumination = true;
+          enterTime = nextIlluminationTime.getTime();
+        }
+      } else {
+        if (enterIllumination && enterTime !== null) {
+          exitInfo = {
+            type: "Exit",
+            time: formattedTime,
+            longitude: longitude,
+            latitude: latitude,
+          };
+
+          if (
+            nextIlluminationTime.getTime() - enterTime >=
+            MIN_CYCLE_DURATION
+          ) {
+            nextIlluminations.push([enterInfo, exitInfo]);
+          }
+          enterIllumination = null;
+          enterTime = null;
```

Comment:
Could you add some comments describing what's happening here?

Reply:
Small nitpick, can you change the 'var' declarations to 'let'

Reply:
sir yes sir